producer/writer/message.go

```diff
@@ -33,47 +32,52 @@ type message struct {
 
 	pb           msgpb.Message
 	meta         metadata
-	retryAtNanos *atomic.Int64
-	retried      *atomic.Int64
-	isAcked      *atomic.Bool
+	retryAtNanos int64
```

Comment:
Is this only accessed from one goroutine?

Reply:
Yep

Reply:
Cool, do you still need `isAcked` to be atomic then? Or is it that some of these are accessed only from one goroutine, and others could be accessed concurrently? If the latter, would recommend adding short comments next to these field to explain their access patterns to save future readers any confusion around this.

Reply:
Yeah modified the comment